### PR TITLE
Feature/dynamic coin list

### LINKED ISSUE #13: Implement Dynamic Build of Coin List
Coin list may be dynamically generated from API https://www.cryptocompare.com/api/data/coinlist/

This may however bring in interesting considerations with coin icons and will build in a dependancy to external API to the app's load.

### PR DESCRIPTION
**Pull request by chrislennon**
closes #13 

- Implements a dynamically generated list of coins (from cryptocompare API)
- Implements searchable dropdown list of non 'popular' coins (i.e. coins with no icon yet)
- Maintains existing functionality and provides it to the dynamically added coins
- Removes coins with no icon from the main overview

@BJJLangedijk you wanna test/check? Or you wanna bitch post merge? :neckbeard:

### CODE REVIEW

File: js/crypto.js [hunk 1]
```diff
@@ -156,11 +162,117 @@ function generateJSON(el) {
     el.setAttribute('download', 'Crypto-Touchbar-App-' + selectedFiatObj.ticker + '.json');
 }
 
+function addCustomCoin(customCoin) {
```
Comment (BJJLangedijk) [inline js/crypto.js]: I think I've seen this coin creation stuff somewhere else in the file, can't we make a generic function for it?

File: js/crypto.js [hunk 2]
```diff
@@ -156,11 +162,117 @@ function generateJSON(el) {
...
+    iconCanv.id = customCoin.value;
+    document.getElementById('canvas-area').appendChild(iconCanv);
+
+    // Initialise jscolor on new element
+    jscolor.installByClassName('jscolor');
+
+}
+
+function removeCustomCoin(event) {
+    console.log(event);
```
Comment (BJJLangedijk) [inline js/crypto.js]: Probably should remove this

File: js/crypto.js [hunk 3]
```diff
@@ -156,11 +162,117 @@ function generateJSON(el) {
...
+    const cryptoSelector = document.createElement('div'),
+    element = document.createElement('input'),
+    text = document.createElement('label'),
+    colour = document.createElement('button'),
+    icon = document.createElement('img'),
+    iconCanv = document.createElement('canvas');
+
+    element.type = 'checkbox';
+    element.value = customCoin.value;
+    element.addEventListener('change', addCrypto, false);
```
Comment (BJJLangedijk) [inline js/crypto.js]: Is the `false` needed?

File: js/crypto.js [hunk 4]
```diff
@@ -224,4 +337,14 @@ function loadData() {
...
+
+    // events for on change of searchbox input
+
+    document.getElementById('dynamic-coinlist').addEventListener('addItem', function(event) {
+        addCustomCoin(event.detail);
+      }, false);
+    
+    document.getElementById('dynamic-coinlist').addEventListener('removeItem', function(event) {
+        removeCustomCoin(event.detail);
+      }, false);
```
Comment (BJJLangedijk) [inline js/crypto.js]: Could probably make a var out of `document.getElementById('dynamic-coinlist')`, and `false` needed?

File: js/crypto.js (hunk 1 above)
Comment (chrislennon) [reply js/crypto.js]: Probably yeah, i was being lazy but fair point

File: js/crypto.js (hunk 2 above)
Comment (chrislennon) [reply js/crypto.js]: nothing to see here!

File: js/crypto.js (hunk 3 above)
Comment (chrislennon) [reply js/crypto.js]: think this is a hangover from misunderstanding how to pass in parameters to the event. now handled by `element.someParam = value` . Consider it gone.

File: js/crypto.js (hunk 4 above)
Comment (chrislennon) [reply js/crypto.js]: _could_ possibly do lots of things, but this seems a sensible suggestion ;)

Comment (BJJLangedijk) [reply js/crypto.js]: Also,doesn't this work:

document.getElementById('dynamic-coinlist').addEventListener('removeItem', removeCustomCoin);

And then just selecting the `detail` from the `event` being passed along to the function

**Comment by chrislennon** [comment]
changes made, 5dfdb87 included a bit of tidy up as well, may do more like that going forward but this was a good start

**Pull request merged**